packages/app/metadata/actions.graphql

```diff
@@ -9,61 +9,132 @@ enum ErrorType {
   VALIDATION_FAILED,
 }
 
-# getAlertmanager
+# NOTE: Unable to use nested response types in Hasura Actions, so using a flat response for now.
+#       See also https://github.com/hasura/graphql-engine/issues/4796
+# TODO: Is this fixed by switching to Remote Schemas instead?
+type StatusResponse {
+  # False if the update failed for any reason, see error_* fields for details
+  success: Boolean!
+
+  # Must be set if success=false
+  error_type: ErrorType
+  # Must be set if success=false
+  error_message: String
+  # Optional if success=false, message returned by underlying service
+  error_raw_response: String
+}
+
+# getAlertmanager/updateAlertmanager
 
 type Query {
   getAlertmanager(tenant_id: String!): Alertmanager
 }
+type Mutation {
+  updateAlertmanager(tenant_id: String!, input: AlertmanagerInput): StatusResponse
+}
 
 type Alertmanager {
   tenant_id: String!
   config: String
   online: Boolean!
 }
 
-# updateAlertmanager
-
-type Mutation {
-  updateAlertmanager(tenant_id: String!, input: AlertmanagerInput): AlertmanagerUpdateResponse
-}
-
 input AlertmanagerInput {
   config: String!
 }
 
-# NOTE: Unable to use nested response types in Hasura Actions, so using a flat response for now.
-#       See also https://github.com/hasura/graphql-engine/issues/4796
-# TODO: Is this fixed by switching to Remote Schemas instead?
-type AlertmanagerUpdateResponse {
-  # False if the update failed for any reason, see error_* fields for details
-  success: Boolean!
+# alert rules access
+# within each tenant, rules are under this hierarchy: namespace -> rulegroup -> rules
 
-  # Must be set if success=false
-  error_type: ErrorType
-  # Must be set if success=false
-  error_message: String
-  # Optional if success=false, message returned by underlying service
-  error_raw_response: String
+# Returns all rules across all namespaces/rulegroups. Example:
+type Query {
```

Comment:
This is the "meat" the the PR: the added GraphQL queries themselves.

One complication I had was that the RuleGroups are submitted with the name in the YAML payload, so I tried to make the structure work around that. See also https://cortexmetrics.io/docs/api/#set-rule-group